hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/SparkLazyInsertIterable.java

```diff
@@ -34,14 +35,18 @@
 
 public class SparkLazyInsertIterable<T extends HoodieRecordPayload> extends HoodieLazyInsertIterable<T> {
 
+  private boolean addMetadataFields;
```

Comment:
Could you explain this change please ?

Reply:
With regular bulk insert, RDD only has user specified columns. With bulk insert based clustering, RDD also has hoodie internal columns. So, I am adding this flag to make it work for both cases. Let me know if you want me to reorganize this  differently.